### PR TITLE
Hotfix/ath rest impl dep and release jdl

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- Prevent "target" from getting deleted unintentionally when mvn release:prepare and release:perform are executed

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
won't know if this works until we try it - but need to have a review from @vivek and probably @kzantow

File: blueocean-core-js/gulpfile.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const fs = require('fs');
 // Options, src/dest folders, etc
 
 const config = {
-    clean: ["dist", "target"],
+    clean: ["dist"],
```
Comment (vivek) [inline blueocean-core-js/gulpfile.js]: Is there side effect of doing this? generated jar file has js code in it, can dist and target can ever get out of sync?

File: pom.xml [hunk 2]
```diff
@@ -1072,18 +1072,6 @@
...
 
-                            <execution>
-                                <phase>initialize</phase>
-                                <id>gulp clean</id>
-                                <goals>
-                                    <goal>npm</goal>
-                                </goals>
-                                <configuration>
-                                    <!-- The package.json must define an "mvnclean" script -->
-                                    <arguments>run mvnclean</arguments>
```
Comment (vivek) [inline pom.xml]: Hmm, intent of this was to remove jar file during gulp clean. Which sounds ok considering generated jar file includes result of gulp build or maybe I misunderstand this process?

File: blueocean-core-js/gulpfile.js (hunk 1 above)
Comment (michaelneale) [reply blueocean-core-js/gulpfile.js]: I believe people do a clean anyway...

File: pom.xml (hunk 2 above)
Comment (kzantow) [reply pom.xml]: ehh. not sure why this was here, I didn't add it, did I? :)

**Comment by michaelneale** [comment]
so current progress: 

@vivek is going to try to do a dryRun locally with master, check can repro the release failure
Will then try it with just the change in the parent pom phase to "clean" for the gulp clean block
if that works, will open a PR just with that one line change... 

failing that, we may fall back to this.

**Comment by cliffmeyers** [comment]
@michaelneale @vivek I think you will have problems firing the gulp clean in the clean phase, because it fires before initialize and therefore node and npm won't have been installed yet.

**Comment by michaelneale** [comment]
@cliffmeyers I would think init is first of all based on that list.. but who knows.

**Comment by cliffmeyers** [comment]
https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#Lifecycle_Reference

**Comment by michaelneale** [comment]
@cliffmeyers still not clear to me - do we even need that clean in the parent pom - perhaps if that was removed would be enough?

File: pom.xml (hunk 2 above)
Comment (cliffmeyers) [reply pom.xml]: The intent wasn't to remove the .jar file, I was trying to clear out the stuff in `target` that was copied by maven-resources-plugin. Deleting all of `target` was clearly a bad idea, I should have individually cleaned out the folders copied it, just more of a maintenance headache.

**Comment by vivek** [comment]
Tried experimenting with dryRun to test alternate proposal from @kzantow. dryRun doesn't do anything meaningful for release:perform. We are going to merge this PR and try a release after successful master run.

**Pull request merged**